docs/sources/flow/reference/components/prometheus.exporter.azure.md

```diff
@@ -0,0 +1,147 @@
+| `metric_name_template`        | `string`       | Metric template used to expose the metrics.                          | `"azure_{type}_{metric}_{aggregation}_{unit}"`                                | no       |
+| `metric_help_template`        | `string`       | Description of the metric.                                           | `"Azure metric {metric} for {type} with aggregation {aggregation} as {unit}"` | no       |
+
+The list of available `resource_type` values and their corresponding `metrics` can be found in [Azure Monitor essentials][].
+
+The `resource_graph_query_filter` can be embedded into a template query of the form `Resources | where type =~ "<resource_type>" <resource_graph_query_filter> | project id, tags`.
+
+Valid values for `metric_aggregations` are `minimum`, `maximum`, `average`, `total`, and `count`. If no aggregation is specified, the value is retrieved from the metric. For example the aggregation value of the metric `Availability` in [Microsoft.ClassicStorage/storageAccounts](https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-classicstorage-storageaccounts-metrics) is `average`.
+
+Every metric has its own set of dimensions. For example the dimensions for the metric `Availability` in [Microsoft.ClassicStorage/storageAccounts](https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-classicstorage-storageaccounts-metrics) are `GeoType`, `ApiName`, and `Authentication`. If a single dimension is requested it will have the name `dimension`. If multiple dimensions are requested they will have the name `dimension<dimension_name>`.
+
+Tags in `included_resource_tags` will be added as labels with the name `tag_<tag_name>`.
+
+Valid values for `azure_cloud_environment` are `azurecloud`, `azurechinacloud`, `azuregovernmentcloud` and `azurepprivatecloud`.
+
+[Kusto query]: https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/
+[Azure Monitor essentials]: https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/metrics-supported
+[ISO8601 Duration]: https://en.wikipedia.org/wiki/ISO_8601#Durations
+
+## Exported fields
+
+{{< docs/shared lookup="flow/reference/components/exporter-component-exports.md" source="agent" version="<AGENT VERSION>" >}}
+
+## Component health
+
+`prometheus.exporter.azure` is only reported as unhealthy if given
+an invalid configuration. In those cases, exported fields retain their last healthy values.
+
+## Debug information
+
+`prometheus.exporter.azure` does not expose any component-specific
+debug information.
+
+## Debug metrics
+
+`prometheus.exporter.azure` does not expose any component-specific
+debug metrics.
+
+## Examples
+
+```river
+prometheus.exporter.azure "example" {
+	subscriptions    = ["179c4f30-ebd8-489e-92bc-fb64588dadb3"]
```

Comment:
Can we parameterize this as well in the example?

Reply:
I think we should parameterize things that would make it easier for users to run on their own environments. Here, people may choose to use our metrics set, but they will definitely have to alter their subscription, that's why I suggested this.